stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/message/MessageSendingService.kt

```diff
@@ -65,15 +65,20 @@ internal class MessageSendingService(
     }
 
     internal suspend fun sendMessage(message: Message): Result<Message> {
-        return if (domainImpl.online.value) {
-            return if (message.hasPendingAttachments()) {
-                waitForAttachmentsToBeSent(message)
-            } else {
-                doSend(message)
+        return when {
+            domainImpl.online.value ->
+                if (message.hasPendingAttachments()) {
+                    waitForAttachmentsToBeSent(message)
+                } else {
+                    doSend(message)
+                }
+
+            message.hasPendingAttachments() -> waitForAttachmentsToBeSent(message)
```

Comment:
Why do you want to wait for attachments if there is no online?

Reply:
Makes sense, there's no need to wait. It only enqueues the message